File: hsg9-tinh/2023-2024-haiduong/banhtrungthu/banhtrungthu.cpp
#include <iostream>

#define input_file "banhtrungthu.inp"
#define output_file "banhtrungthu.out"

using namespace std;

int m, n;
int result;


void input()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    freopen(input_file, "r", stdin);

    cin >> m;
    cin >> n;
}


void process()
{
    // Số lượng giá trị k chia hết cho m mà không vượt quá n
    int count_n_m = n / m;

    // Số lượng giá trị k +  1 chia hết cho m mà không vượt quá n
    int count_n_1_m = (n + 1) / m;
    
    // Số lượng cần tìm
    result = count_n_m + count_n_1_m;
}


void output()
{
    freopen(output_file, "w", stdout);

    cout << result;
}


int main()
{
    input();
    process();
    output();

    return 0;
}

File: hsg9-tinh/2023-2024-haiduong/banhtrungthu/banhtrungthu.py
import os
import sys

input_file = os.path.join(sys.path[0], 'banhtrungthu.inp')
output_file = os.path.join(sys.path[0], 'banhtrungthu.out')

m = n = 0
result = 0


def input():
    global m, n

    with open(input_file, 'r') as f:
        m = int(f.readline())
        n = int(f.readline())


def process():
    global n, m, result

    # Số lượng giá trị k chia hết cho m mà không vượt quá n
    count_n_m = n // m

    # Số lượng giá trị k + 1 chia hết cho m mà không vượt quá n
    count_n_1_m = (n + 1) // m

    # Số lượng cần tìm
    result = count_n_m + count_n_1_m


def output():
    global result

    with open(output_file, 'w') as f:
        f.write(str(result))


if __name__ == '__main__':
    input()
    process()
    output()
